### PR TITLE
[UI] Added helm connection Wizard

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

- [X] Added UI for helm connection wizard on path `/connect/helm/new`
- [X] Migrated UI from MUI v5 to v4
- [x] Support All new APIs

**Screenshot**
Screenshot 2023-09-29 at 6 21 43 PM

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/Connect/styles.js
Comment (Yashsharma1911) [inline ui/components/Connect/styles.js:1]: Not using this file though keeping it until it is required and not error comes in UI

**Review by Yashsharma1911 — COMMENTED** [review]
Removing these

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+  const schema = {
+    properties: {
+      selectedHelmRepos: {
+        description: 'Select one or more Helm charts from the available options',
+        items: {
+          enum: availableRepos,
+          type: 'string',
+        },
+        minItems: 1,
+        title: 'Select one or more helm chart',
```
Comment (leecalcote) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:201]:
```suggestion
        title: 'Select one or more Helm Charts',
```

File: ui/assets/icons/FolderIcon.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    width="1.5rem"
+    height="1.5rem"
+    viewBox="0 0 31 30"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <g clipPath="url(#a)">
+      <path
+        d="M28.625 3.75h-11.25V1.875A1.874 1.874 0 0 0 15.5 0H2.375C1.339 0 .5.84.5 1.875v26.25c0 .473.18.9.469 1.23v.176h.177c.33.289.756.469 1.229.469h26.25c.473 0 .9-.18 1.23-.469h.176v-.177a1.86 1.86 0 0 0 .469-1.229v-22.5a1.874 1.874 0 0 0-1.875-1.875Zm-5.156 17.813H7.53a.468.468 0 1 1 0-.938h15.94a.468.468 0 1 1 0 .938Zm0-2.813H7.53a.468.468 0 1 1 0-.938h15.94a.468.468 0 1 1 0 .938ZM7.062 15.469c0-.26.21-.469.47-.469h7.5a.468.468 0 1 1 0 .938h-7.5a.468.468 0 0 1-.47-.47Zm22.5-8.906H1.438V1.875c0-.518.42-.938.938-.938H15.5c.517 0 .938.42.938.938v2.813h12.187c.517 0 .938.42.938.937v.938Z"
+        fill="#3C494F"
```
Comment (leecalcote) [inline ui/assets/icons/FolderIcon.js:14]:
```suggestion
        fill={fill]
```

File: ui/assets/icons/FolderIcon.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+const FolderIcon = ({ width, height, fill, style = {} }) => (
```
Comment (leecalcote) [inline ui/assets/icons/FolderIcon.js:3]:
```suggestion
const FolderIcon = ({ width, height, fill="#3C494F", style = {} }) => (
```

File: ui/components/Connect/helpers/common.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Verify connection status
+ * If connection already exist response will be connection object
+ * If connection is not exist response will be 200 empty
+ * 
+ * @param {string} connectionKind - Kind of connection e.g (helm, github) pass only managed connections
+*/
+export const VerifyConnectionKind = async (connectionKind) => {
+    return await axios.post(`/api/integrations/${connectionKind}/status`);
```
Comment (leecalcote) [inline ui/components/Connect/helpers/common.js:9]: @aabidsofi19 are we getting rid of axios?

**Review by leecalcote — CHANGES_REQUESTED** [review]
This branch doesn't build.

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+      btnText={isConnected ? 'Update' : 'Connect'}
+    >
+      {isSchemaFetched && (
+        <RJSFWrapper
+          key="connect-helm-repo-rjsf-form"
+          formData={formData}
+          onChange={handleChange}
+          uiSchema={schema.uiSchema}
+          jsonSchema={schema.rjsfSchema}
+          liveValidate={false}
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:106]: @Yashsharma1911 By enabling this I think we can get rid of the behaviour that user needs to submit before RJSF could validate.

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+          formRef={formRef}
+          // setErrors={setErrors}
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:108]: Missed by mistake?

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+            }
+          >
+            <AlertTitle style={{ fontSize: '1.4rem' }}>{name}</AlertTitle>
+            {isSuccess ? (
+              <div>
+                <p>{`Design ${name} created`}</p>
+              </div>
+            ) : (
+              <div>
+                <p>{`Unable to import ${name}.yaml`}</p>
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:290]:
```suggestion
                <p>{`Unable to import <name of chart>`}</p>
```
@Yashsharma1911 this should contain the name of chart/<and the version of the chart>

File: ui/components/Connect/connections/helm-connect/constants.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  tips: "Meshery is optimized for GitOps-centric workflows. Connect your  GitHub repositories for a full experience.",
+  btnText: "Connect",
+  cancel: true
+};
+
+export const HELM_INTEGRATION_ENDPOINT = "/api/integrations/github";
+
+export const SelectRepositoryContent = {
+  title: "Select Helm Repository",
+  tips: "Meshery Cloud is optimized for codebases that are actively undergoing development.",
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/constants.js:57]:
```suggestion
  tips: "Meshery is optimized for codebases that are actively undergoing development.",
```

File: ui/components/Connect/connections/helm-connect/utils.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+export const postHelmInstall = async (body, installationId) => {
+  const res = await axios.post(
+    process.env.API_ENDPOINT_PREFIX +
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/utils.js:5]: Not required here.

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 10]
```diff
@@ -0,0 +1,369 @@
...
+  const formRef = React.createRef();
+  const formStateRef = React.createRef();
+  const [schema, setSchema] = React.useState({});
+  const [isSchemaFetched, setIsSchemaFetched] = React.useState(false);
+  const [formData, setFormData] = React.useState({});
+
+  const handleCallback = () => {
+    if (formRef.current && formRef.current.validateForm()) {
+      // Check if the URL is valid via an API 
+      axios
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:27]: Axios?

We don’t use axios in Meshery UI but fetchAPI.

Comment (MUzairS15) [reply ui/components/Connect/connections/helm-connect/StepperContent.js:27]: Have we started using RTK in MesheryUI?
RTK by default uses fetch API, so if we have not started you can initiate usage of RTK , you can do that (in this PR or separately)

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: ui/components/Connect/helpers/common.js (hunk 4 above)
Comment (aabidsofi19) [reply ui/components/Connect/helpers/common.js:9]: yup , we are . @Yashsharma1911 you can create a rtk endpoint(s) for connections for now ( till we start using the codegen)

**Comment by saurabh100ni** [comment]
Let's discuss this on the Meshery dev call tomorrow,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

File: ui/components/Connect/helpers/common.js (hunk 4 above)
Comment (Yashsharma1911) [reply ui/components/Connect/helpers/common.js]: Removed, RTK used

File: ui/components/Connect/connections/helm-connect/StepperContent.js (hunk 5 above)
Comment (Yashsharma1911) [reply ui/components/Connect/connections/helm-connect/StepperContent.js:153]: Yeah though this gives error, before even user type on field, although I added debounce on onChange, which works same

File: ui/components/Connect/connections/helm-connect/StepperContent.js (hunk 7 above)
Comment (Yashsharma1911) [reply ui/components/Connect/connections/helm-connect/StepperContent.js]: fixed

File: ui/components/Connect/connections/helm-connect/StepperContent.js (hunk 6 above)
Comment (Yashsharma1911) [reply ui/components/Connect/connections/helm-connect/StepperContent.js]: No it was intentional, though we can use extraErrors prop of RJSF to do async validation

File: ui/components/Connect/connections/helm-connect/utils.js (hunk 9 above)
Comment (Yashsharma1911) [reply ui/components/Connect/connections/helm-connect/utils.js]: removed

File: ui/components/Connect/connections/helm-connect/StepperContent.js (hunk 10 above)
Comment (Yashsharma1911) [reply ui/components/Connect/connections/helm-connect/StepperContent.js]: Yes we are using RTK now I have created all RTK APIs

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+  const [verifyConnectionURL] = useVerifyConnectionURLMutation();
+  const { data: schema, isSuccess: isSchemaFetched } = useGetSchemaQuery({
+    schemaName: 'helmRepo',
+  });
+  const { data: connectionStatusData, isSuccess: isConnectionStatusDataFetched } =
+    useGetConnectionStatusQuery('helm');
+
+  const handleCallback = async () => {
+    setExtraErrors({});
+    // replace !isUrlValid to isUrlValid after testing
```
Comment (Yashsharma1911) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:47]:
```suggestion
```

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 12]
```diff
@@ -0,0 +1,388 @@
...
+      }
+    }
+  }, [isConnectionStatusDataFetched]);
+
+  return (
+    <StepperContent
+      {...ConnectAppContent}
+      handleCallback={handleCallback}
+      cancelCallback={cancelCallback}
+      // disabled={Object.keys(formData).length !== 3 || !isUrlValid} //uncomment this
```
Comment (Yashsharma1911) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:142]:
```suggestion
      disabled={Object.keys(formData).length !== 3 || !isUrlValid}
```

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 13]
```diff
@@ -0,0 +1,388 @@
...
+      disabled={Object.keys(formData).length !== 3} // remove this after testing
```
Comment (Yashsharma1911) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:143]:
```suggestion
```

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 14]
```diff
@@ -0,0 +1,386 @@
...
+        })
+        .catch((error) => {
+          // TODO: use notify here
+          console.error('Error coming when fetching connection details', error);
+        });
+    }
+  };
+
+  const cancelCallback = () => {
+    Router.push('/dashboard');
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:90]: Redirect/stay at the connection page itself.

File: ui/components/Connect/connections/helm-connect/constants.js [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+export const ConnectAppContent = {
+  title: 'Connecting your Helm Repository',
+  subtitle:
+    'Integrate Meshery seamlessly with your existing Helm-based process. Connect Meshery to one or more Helm repositories and import one or all Helm charts from your repo.',
+  tips: 'Meshery is optimized for GitOps-centric workflows. Connect your  GitHub repositories for a full experience.',
+  btnText: 'Connect',
+  cancel: true,
+};
+
+export const HELM_INTEGRATION_ENDPOINT = '/api/integrations/github';
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/constants.js:44]: Please check the endpoint.

File: ui/components/Connect/connections/helm-connect/constants.js [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+  'GitOps is more fun with friends. Invite one today.',
+  'Meshery works with multiple Kubernetes clusters. Connect each of yours by uploading your kubeconfig and letting Meshery do the rest.',
+  'Import your Docker Compose, Helm Chart, and Kubernetes mainfests. Let Meshery visualize and deploy them.',
+];
+
+export const ConnectAppContent = {
+  title: 'Connecting your Helm Repository',
+  subtitle:
+    'Integrate Meshery seamlessly with your existing Helm-based process. Connect Meshery to one or more Helm repositories and import one or all Helm charts from your repo.',
+  tips: 'Meshery is optimized for GitOps-centric workflows. Connect your  GitHub repositories for a full experience.',
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/constants.js:39]:
```suggestion
  tips: 'Meshery is optimized for GitOps-centric workflows. Connect your  Helm repositories for a full experience.',
```

File: ui/components/Connect/connections/helm-connect/StepperContent.js [hunk 17]
```diff
@@ -0,0 +1,386 @@
...
+          }
+        >
+          <AlertTitle style={{ fontSize: '1.4rem' }}>{name}</AlertTitle>
+          {isSuccess ? (
+            <div>
+              <p>{`Design ${name} created`}</p>
+            </div>
+          ) : (
+            <div>
+              <p>{`Unable to import ${name}.yaml`}</p>
```
Comment (MUzairS15) [inline ui/components/Connect/connections/helm-connect/StepperContent.js:304]: Unable to import `repo/chartname`

**Review by MUzairS15 — APPROVED** [review]
LGTM

File: ui/components/Connect/connections/helm-connect/constants.js (hunk 15 above)
Comment (Yashsharma1911) [reply ui/components/Connect/connections/helm-connect/constants.js:44]: Ohh I removed it didn't push the changes, updating it

**Comment by Yashsharma1911** [comment]
@MUzairS15 I'm merging this PR, later I'll test this PR with your APIs and if some change is required I'll raise a new PR with required changes

**Pull request merged**